qiita_client/qiita_client.py

```diff
@@ -28,18 +28,24 @@ class ArtifactInfo(object):
         Qiita's artifact type
     files : list of (str, str)
         The list of (filepath, Qiita's filepath type) that form the artifact
+    archive : dict of {str: str}, optional
+        A dict of features and their values to store. Format: {feature: values}
     """
-    def __init__(self, output_name, artifact_type, files):
+    def __init__(self, output_name, artifact_type, files, archive={}):
         self.output_name = output_name
         self.artifact_type = artifact_type
         self.files = files
+        self.archive = archive
 
     def __eq__(self, other):
         if type(self) != type(other):
             return False
         if self.output_name != other.output_name or \
                 self.artifact_type != other.artifact_type or \
-                set(self.files) != set(other.files):
+                set(self.files) != set(other.files) or \
+                self.archive != other.archive or \
```

Comment:
This is redundant. Since archive is a dictionary it already takes into account the key values:

```python
In [1]: d1 = {'a': 1, 'b': 2}

In [2]: d2 = {'a': 1, 'b': 3}

In [3]: d1 == d2
Out[3]: False

In [4]: d2['b'] = 2

In [5]: d1 == d2
Out[5]: True
```